rustls/src/msgs/deframer.rs

```diff
@@ -87,9 +91,22 @@ impl MessageDeframer {
                 }
             };
 
-            // Return CCS messages immediately without decrypting.
+            // Return CCS messages and early plaintext alerts immediately without decrypting.
             let end = start + rd.used();
-            if m.typ == ContentType::ChangeCipherSpec && self.joining_hs.is_none() {
+            let type_is_ccs = m.typ == ContentType::ChangeCipherSpec;
+            let has_decrypted = record_layer.has_decrypted();
+            let type_is_alert = m.typ == ContentType::Alert;
+            let payload_is_alert = m.payload.0.len() <= 2;
+            let version_is_tls13 = matches!(negotiated_version, Some(ProtocolVersion::TLSv1_3));
+            // We consider a message an early plaintext alert if-and-only-if:
+            //  * We haven't decrypted any messages from the peer yet.
+            //  * The message content type is Alert.
+            //  * The payload size is indicative of a plaintext alert message.
+            //  * The negotiated protocol version is TLS 1.3.
+            // In all other circumstances a plaintext alert is not expected.
+            let early_plaintext_alert =
```

Comment:
Style nit: have you tried inlining the values from above in this statement? That would save a lot of bindings, and maybe there's room for comments on each sub-expression's line?

Reply:
Sure, I'm open to that. 

I had left these individual bindings in the initial push because I found them helpful when verifying the test coverage. When it was all one big boolean expression it was more difficult to determine _why_ the decision to not try and treat the message as plaintext was reached - was it the payload size? The protocol version? Etc.

I think with the test coverage established that's not a super great reason to keep it as verbose as it is. I'll run with `match` suggestion from Ctz.

Reply:
Implemented the `match` suggestion.